openssl/src/pkey.rs

```diff
@@ -524,6 +524,27 @@ impl PKey<Private> {
         ffi::d2i_AutoPrivateKey
     }
 
+    /// Deserializes a DER-formatted PKCS#8 unencrypted private key.
+    ///
+    /// This method is mainly for interoperability reasons. Encrypted keyfiles should be preferred.
+    pub fn private_key_from_pkcs8(
+        der: &[u8],
+    ) -> Result<PKey<Private>, ErrorStack>
+    {
+        unsafe {
+            ffi::init();
+            let bio = MemBioSlice::new(der)?;
+            let p8inf = cvt_p(ffi::d2i_PKCS8_PRIV_KEY_INFO_bio(
```

Comment:
Oh - one last bit of cleanup. This can use `d2i_PKCS8_PRIV_KEY_INFO` to avoid having to make the bio.

Reply:
Fixed.